### ISSUE TITLE
Register for announce list during account creation is broken

### ISSUE THREAD

**Issue by jvendetti** [ncbo/bioportal_web_ui#60]
Steps to reproduce:

1. Navigate to the [account creation page](https://bioportal.bioontology.org/accounts/new) in BioPortal.
2. Fill in the form fields and check the box to "Register for the BioPortal announcements email list".
3. Click the Create button, and notice that no confirmation email is ever sent to complete the subscription process.

I've confirmed that sending email isn't the problem (I can successfully receive an email for a password reminder). Also confirmed via the Mailman interface that email addresses don't get added to the list.

On a related note, the UI says that you are requesting a subscription to the announce list, but the production config file is actually [pointing to the list address of bioontology-support](https://github.com/ncbo/bioportal_config/blob/master/bioportal_web_ui/config/bioportal_config_production.rb#L12). This should perhaps be bioontology-announce instead?

**Comment by graybeal** [comment]
Yes, we should be offering the ability to register for both announce and support there. And making clear what they are. (I keep waffling on announce between low frequence community announcements, and lower-frequency BioPortal announcements.)
